src/Features/Core/Portable/ConvertIfToSwitch/AbstractConvertIfToSwitchCodeRefactoringProvider.Analyzer.cs

```diff
@@ -433,8 +433,18 @@ private static bool IsConstant(IOperation operation)
 
             private bool CheckTargetExpression(IOperation operation)
             {
-                if (operation is IConversionOperation { IsImplicit: false } op)
+                if (operation is IConversionOperation op)
                 {
+                    // We can't have implicit conversions because switch arms don't do them.
+                    // Putting an explicit cast on the arm doesn't have the same semantics. For example:
+                    //    if (c == 123)
+                    // is the same as
+                    //    if ((int)c == 123)
+                    // but putting the cast on the switch arm would be equivalient to
+                    //    if (c == (char)123)
+                    if (op.IsImplicit)
+                        return false;
```

Comment:
so i'm totally ok with this.  We *Could* consider adding a comment that if all the branches were, say, integers, that we could potentially cast the switch-expression upwards to that.    but for now it seems acceptable that we don't have a great idiomatic thing to transform this to, so we bail on it.